### ISSUE TITLE
Warning: TempContainer trying to remove non-child actor...

### ISSUE THREAD

**Issue by Aaronontheweb** [akkadotnet/akka.net#1448]
I see this continuously inside the logging center for the multinode testrunner and I think it's related to `FutureActorRef` and `Ask`, but either way it's disconcerting and I want to understand the circumstances under which this warning is raised. Might not be a big deal, but we should investigate.

```
[WARNING][11/19/2015 11:27:19 PM][Thread 0005][LocalActorRefProvider(akka://TestRunnerLogging)] akka://TestRunnerLogging/temp trying to remove non-child pb
```

**Comment by rogeralsing** [comment]
There are two scenarios where this could potentially happen.

1. Some code tries to unregister a non existing temp actorref. e.g. passing in the incorrect name
2. Some code for some reasons tries to remove the same child twice (or more than one time)

My guess would be that there is some code in the `Ask` mechanics that might do the latter, e.g. unregister temp in some combination of complete and failure of tasks.

**Comment by rogeralsing** [comment]
The `VirtualPathContainer` which logs this is also used for hosting the remote deployed actors in Akka.Remote. 
`class RemoteSystemDaemon : VirtualPathContainer`

**Comment by Aaronontheweb** [comment]
> Some code for some reasons tries to remove the same child twice (or more than one time)

I suspect that's what is happening

**Comment by Aaronontheweb** [comment]
Closed in #2422
